### PR TITLE
Add support for Azure 2.0.0

### PR DESCRIPTION
**Pull request by trstringer**
This pull request is a superset of @lmazuel's PR #24710.

This PR works off of @lmazuel's work on adding azure==2.0.0 support to Ansible. As it current stands, the following playbook is what I used to test out the changes (with the current lack of integration testing, which is also currently undergoing development):

```
---
- name: testing azure 2.0.0 sdk
  hosts: localhost

  tasks:
    - name: create resource group
      azure_rm_resourcegroup:
        name: ansiblefinal-rg
        location: eastus

    - name: create virtual network
      azure_rm_virtualnetwork:
        resource_group: ansiblefinal-rg
        name: ansiblefinalvnet
        address_prefixes_cidr:
          - "11.1.0.0/16"

    - name: create subnet
      azure_rm_subnet:
        resource_group: ansiblefinal-rg
        virtual_network: ansiblefinalvnet
        address_prefix_cidr: "11.1.0.0/16"
        name: ansiblesubnetfinal

    - name: create nsg
      azure_rm_securitygroup:
        resource_group: ansiblefinal-rg
        name: ansiblefinalnsg
        rules:
          - name: allowssh
            protocol: Tcp
            destination_port_range: 22
            access: Allow
            priority: 100
            direction: Inbound

    - name: create pip
      azure_rm_publicipaddress:
        name: ansiblefinalpip
        resource_group: ansiblefinal-rg

    - name: create nic
      azure_rm_networkinterface:
        name: ansiblefinalnic
        resource_group: ansiblefinal-rg
        public_ip_address: ansiblefinalpip
        subnet: ansiblesubnetfinal
        virtual_network: ansiblefinalvnet
        security_group: ansiblefinalnsg

    - name: create storage account
      azure_rm_storageaccount:
        name: ansiblefinalstor2
        resource_group: ansiblefinal-rg
        account_type: Standard_LRS

    - name: get ssh key
      command: cat ~/.ssh/id_rsa.pub
      register: sshkey

    - name: create vm
      azure_rm_virtualmachine:
        resource_group: ansiblefinal-rg
        name: ansiblefinalvm
        vm_size: Standard_D1
        storage_account: ansiblefinalstor2
        admin_username: adminUser
        ssh_password_enabled: false
        ssh_public_keys:
          - path: /home/adminUser/.ssh/authorized_keys
            key_data: '{{ sshkey.stdout }}'
        network_interfaces: ansiblefinalnic
        image:
          offer: CentOS
          publisher: OpenLogic
          sku: '7.3'
          version: latest
```

Please review and let me know if there are any issues that you see, or questions that you have.

### CODE REVIEW

**Comment by trstringer** [comment]
cc @haroldwongms @ozboms @nitzmahone @thaumos

File: lib/ansible/module_utils/azure_rm_common.py [hunk 1]
```diff
@@ -595,7 +634,7 @@ def storage_client(self):
         self.log('Getting storage client...')
         if not self._storage_client:
             self.check_client_version('storage', storage_client_version, AZURE_EXPECTED_VERSIONS['storage_client_version'])
-            self._storage_client = StorageManagementClient(self.azure_credentials, self.subscription_id)
+            self._storage_client = StorageManagementClient(self.azure_credentials, self.subscription_id, base_url=self.base_url)
```
Comment (lmazuel) [inline lib/ansible/module_utils/azure_rm_common.py]: Fix api-version here to "2017-06-01"

File: lib/ansible/module_utils/azure_rm_common.py [hunk 2]
```diff
@@ -604,7 +643,7 @@ def network_client(self):
         self.log('Getting network client')
         if not self._network_client:
             self.check_client_version('network', network_client_version, AZURE_EXPECTED_VERSIONS['network_client_version'])
-            self._network_client = NetworkManagementClient(self.azure_credentials, self.subscription_id)
+            self._network_client = NetworkManagementClient(self.azure_credentials, self.subscription_id, base_url=self.base_url)
```
Comment (lmazuel) [inline lib/ansible/module_utils/azure_rm_common.py]: Fix api-version here to "2017-06-01"

File: lib/ansible/module_utils/azure_rm_common.py [hunk 3]
```diff
@@ -613,14 +652,16 @@ def rm_client(self):
         self.log('Getting resource manager client')
         if not self._resource_client:
             self.check_client_version('resource', resource_client_version, AZURE_EXPECTED_VERSIONS['resource_client_version'])
-            self._resource_client = ResourceManagementClient(self.azure_credentials, self.subscription_id)
+            self._resource_client = ResourceManagementClient(self.azure_credentials,
```
Comment (lmazuel) [inline lib/ansible/module_utils/azure_rm_common.py]: Fix api-version here to "2017-05-10"

File: lib/ansible/module_utils/azure_rm_common.py [hunk 4]
```diff
@@ -613,14 +652,16 @@ def rm_client(self):
...
+                                                             base_url=self.base_url)
         return self._resource_client
 
     @property
     def compute_client(self):
         self.log('Getting compute client')
         if not self._compute_client:
             self.check_client_version('compute', compute_client_version, AZURE_EXPECTED_VERSIONS['compute_client_version'])
-            self._compute_client = ComputeManagementClient(self.azure_credentials, self.subscription_id)
+            self._compute_client = ComputeManagementClient(self.azure_credentials, self.subscription_id, base_url=self.base_url)
```
Comment (lmazuel) [inline lib/ansible/module_utils/azure_rm_common.py]: Fix api-version here to "2017-03-30"

**Review by lmazuel — CHANGES_REQUESTED** [review]
Fix api-version to avoid breaking on package update

**Comment by lmazuel** [comment]
@tstringer what about adding the support for SDK auth file? :)
https://docs.microsoft.com/en-us/python/azure/python-sdk-azure-authenticate?view=azure-python#mgmt-auth-file

File: docs/docsite/rst/guide_azure.rst [hunk 5]
```diff
@@ -8,12 +8,12 @@ Requirements
...
 
 Using the Azure Resource Manager modules requires having `Azure Python SDK <https://github.com/Azure/azure-sdk-for-python>`_
-installed on the host running Ansible. You will need to have == v2.0.0RC5 installed. The simplest way to install the
+installed on the host running Ansible. You will need to have == v2.0.0 installed. The simplest way to install the
 SDK is via pip:
 
 .. code-block:: bash
 
-    $ pip install "azure==2.0.0rc5"
+    $ pip install azure
```
Comment (lmazuel) [inline docs/docsite/rst/guide_azure.rst]: might want to document that "pip install azure-cli" should work and offer the "cli_profile_support"

File: lib/ansible/modules/cloud/azure/azure_rm_resourcegroup.py [hunk 6]
```diff
@@ -233,7 +233,7 @@ def delete_resource_group(self):
     def resources_exist(self):
         found = False
         try:
-            response = self.rm_client.resource_groups.list_resources(self.name)
+            response = self.rm_client.resources.list_by_resource_group(self.name)
```
Comment (lmazuel) [inline lib/ansible/modules/cloud/azure/azure_rm_resourcegroup.py]: we might want to catch "AttributeError" and try the old one `resource_groups.list_resources` to be ready for Azure Stack

File: lib/ansible/module_utils/azure_rm_common.py (hunk 2 above)
Comment (trstringer) [reply lib/ansible/module_utils/azure_rm_common.py]: @lmazuel thanks for the suggestion! Any thoughts on why I'd be getting `APIVersion 2017-06-01 is not available` with NetworkManagementClient (and possibly the other ones as well)?

`pip freeze | grep network` shows me `azure-mgmt-network==1.0.0`

**Comment by mattclay** [comment]
Closing and re-opening to trigger CI.

**Review by ryansb — APPROVED** [review]

**Comment by trstringer** [comment]
@lmazuel thank you for the review! All of the changes have been implemented, except the SDK auth file. I think that's really important, but I think that should be a separate PR in the near future.

**Comment by nitzmahone** [comment]
We should probably talk about the new dep thing- we want to understand the need for it better, because the initial reaction from everyone internally that it's been mentioned to was, "?!"

**Comment by nitzmahone** [comment]
(maybe on IRC Friday or something, unless you want to do it now)

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**